ssl/statem/statem_clnt.c

```diff
@@ -1543,6 +1543,10 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL_CONNECTION *s, PACKET *pkt)
             /* SSLfatal() already called */
             goto err;
         }
+
+        if (SSL_CONNECTION_IS_DTLS13(s)) {
+            SSL_clear_options(ssl, SSL_OP_ENABLE_MIDDLEBOX_COMPAT);
+        }
```

Comment:
Hmm. I'm not sure about this. This is different behaviour to all the other options. With all other options, if the option isn't relevant we leave it set but just ignore it. It seems slightly strange to be actively clearing this one.

Reply:
I was missing the option to force this to off for DTLS13. Maybe there's another way to force it? 

Alternatively I could do `&&  !SSL_CONNECTION_IS_DTLS13(s)` in all places where MIDDLEBOX_COMPAT setting is checked?

Reply:
We actually do a similar thing to this with TLS-1.3 in QUIC. However there it is done for a reason. IMO it would make sense to just ignore MIDDLEBOX_COMPAT if SSL_CONNECTION_IS_DTLS (not just DTLS13).